{
  if (x instanceof EofException) {
    LOG.warn("failed(" + context + ") - "+ EofException.class);
  }
 else {
    LOG.warn("failed(" + context + ")",x);
  }
  cancelTask();
  callback.failed(context,x);
}
